code/modules/spells/spell_types/construct_spells.dm

```diff
@@ -279,4 +312,28 @@
 /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/golem
 	charge_max = 800
 	jaunt_in_type = /obj/effect/temp_visual/dir_setting/cult/phase
-	jaunt_out_type = /obj/effect/temp_visual/dir_setting/cult/phase/out
\ No newline at end of file
+	jaunt_out_type = /obj/effect/temp_visual/dir_setting/cult/phase/out
+
+
+/obj/effect/proc_holder/spell/dumbfire/juggernaut
+	name = "Gauntlet Echo"
+	desc = "Channels energy into your gauntlet - firing its essence forward in a slow-moving but devastating blow."
+	proj_icon_state = "cursehand0"
+	proj_name = "Shadowfist"
+	proj_type = "/obj/effect/proc_holder/spell/targeted/inflict_handler/juggernaut" //IMPORTANT use only subtypes of this
```

Comment:
no dont use a fucking string

Reply:
Q: How can you tell when someone hasn't tested their changes?
A: They don't even WORK

Reply:
the problem with string types, is, well, they work, but if the type name gets deleted or changed, there will be no error, only RUNTIMES

Reply:
yeah strings as types work fine at first but they are a violation of code conventions

Reply:
Q: How can you tell when someone is shitposting?
A: When they claim your actual results are trumped by their expertise in 2d spessman code

I copied the code from Dumbfire projectile spells, which as far as I can tell is the only spelltype we have for this sort of behavior. It's also what we use for stuff like MAGIC MISSILE, one of the most ubiquitous spells we have. I wasn't a fan of learning how these spells work but it also seemed like I would piss people off if I started making all my construct magic into datums.

Reply:
replace `"/obj/effect/proc_holder/spell/targeted/inflict_handler/juggernaut"` with `obj/effect/proc_holder/spell/targeted/inflict_handler/juggernaut`. It should function the same